### PR TITLE
Created FFI for MnemonicWallet

### PR DESCRIPTION
**Pull request by manu0466**
This PR introduces the  **FFI** (Foreign Function Interface) to expose the `crw-wallet` package functionalities to other programming languages.

### CODE REVIEW

**Review by bragaz — CHANGES_REQUESTED** [review]
looks ok to me, some small typos.

**Comment by manu0466** [comment]
@bragaz I have fixed the typos.
During the review i have also noticed that was missing a function to retrieve the public key and so I have added it.

File: packages/crw-wallet/ffi-binding.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ * NULL on error.
+ * The caller must take care of freeing the returned address with the
+ * cstring_free function.
+ * In case of error the error cause can be obtained using the error_message_utf8
+ * function.
+ */
+char* wallet_get_bech32_address(wallet_t *wallet, const char* hrp);
+
+/**
+ * @brief GGets secp256 public key from the wallet.
```
Comment (bragaz) [inline packages/crw-wallet/ffi-binding.h:71]: typo
```suggestion
 * @brief Gets secp256 public key from the wallet.
```

**Review by bragaz — CHANGES_REQUESTED** [review]

File: packages/crw-wallet/ffi-binding.h [hunk 2]
```diff
@@ -0,0 +1,116 @@
+/***
+ * @brief This C header file expose the FFI defined inside the ffi crate.
```
Comment (RiccardoM) [inline packages/crw-wallet/ffi-binding.h:2]:
```suggestion
 * @brief This C header file exposes the FFI defined inside the ffi crate.
```

File: packages/crw-wallet/ffi-binding.h [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+ */
+
+#include <stdint.h>
+
+typedef struct wallet wallet_t;
+
+/**
+ * @brief Struct that represent a signature.
```
Comment (RiccardoM) [inline packages/crw-wallet/ffi-binding.h:10]:
```suggestion
 * @brief Struct that represents a signature.
```

**Review by bragaz — APPROVED** [review]

**Pull request merged**